# HG changeset patch
# User John Donoghue <john.donoghue@ieee.org>
# Date 1374162704 14400
#      Thu Jul 18 11:51:44 2013 -0400
# Node ID 21ed3e2c3be619ee1f67bf7eb9c1c4ce7a69ddda
# Parent  c2e43dd6eb708ff679ea9e1e608d7101667aa5d4
Use HOME environment var for show home folder in files_dock-widget when available

* libgui/src/files-dock-widget.cc
  (files_dock_widget::popdownmenu_home): get HOME env name if available.

diff --git a/libgui/src/files-dock-widget.cc b/libgui/src/files-dock-widget.cc
--- a/libgui/src/files-dock-widget.cc
+++ b/libgui/src/files-dock-widget.cc
@@ -631,17 +631,20 @@ files_dock_widget::notice_settings (cons
   if (_sync_octave_dir)
     display_directory (_octave_dir);  // sync browser to octave dir
 
 }
 
 void
 files_dock_widget::popdownmenu_home (bool)
 {
-  QString dir = QDir::homePath ();
+  QString dir = qgetenv ("HOME");
+  if (dir.isEmpty())
+    dir = QDir::homePath ();
+  
   set_current_directory (dir);
 }
 
 void
 files_dock_widget::popdownmenu_search_dir (bool)
 {
   QString dir = QFileDialog::getExistingDirectory
     (this, tr ("Set directory of file browser"),_file_system_model->rootPath(),
